docs: TECHDOC1891 glossary fix (#4033)
nrfconnect/sdk-nrf: doc/nrf/glossary.rst

   A software application that provides facilities for software development.

Internet Control Message Protocol (ICMP)
   The control protocol of the IP stack that enables the establishment of reachability, routes, and so on.
   This protocol is an integral part of any IP but is unique as it is not a transport protocol to exchange data between hosts.

Internet Protocol version 4 (IPv4)
   The fourth version in the development of the Internet Protocol (IP) Internet, and routes most traffic on the Internet.

Internet Protocol version 6 (IPv6)
   The latest version of the Internet Protocol (IP), the communications protocol that provides an identification and location system for computers on networks and routes traffic across the Internet.

IPv4 address
   A numerical label that is used to identify a network interface of a computer or other network node participating in an IPv4 computer network.

IPv6 address
   An alphanumerical label that is used to identify a network interface of a computer or other network node participating in an IPv6 computer network.

Kconfig
   A kernel configuration language.
   It was initially created for the Linux kernel.

Kconfig file

>    A configuration file provided by an application(usually called prj.conf).

Author: fixed

   It describes build-time configuration options, also called symbols, namely application-specific values for one or more kernel configuration options.

>    It also defines how they’re grouped into menus and sub-menus, and dependencies between them that determine what configurations are valid.

Author: fixed

.. _glossary:

Glossary
########

.. contents::
   :local:
   :depth: 2

Anycast addressing
   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
   This is a one-to-nearest association.

Access Port Protection (APPROTECT)
   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.

API Result
   Indicates success or failure of an API procedure.
   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.

Application
   An implementation example that includes a variety of libraries to implement a specific use case.

Application Programming Interface (API)
   A language and message format used by an application program to communicate with an operating system, application, or other services.

Board
   In Zephyr and |NCS|, a data structure for describing hardware on the board and its configuration files.

Carrier Wave (CW)
   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.

Cat-M1

>    LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.

Reviewer: Should we link to the documents referenced on this page?

Author: I think @OisinFitz was for limiting the number of links in the page.

   A secure system depends on building and maintaining a chain of trust through all the layers in the system.
   Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.

Clone
   A local copy of a remote Git repository obtained with ``git clone``.

Commit
   A Git commit, including a unique SHA and a commit message.

Commit tag
   A tag prepended to the first line of the commit message to ease filtering and identification of particular commit types.

Contribution
   A change to the codebase sent to a remote repository for inclusion.

Development Kit (DK)
   A hardware development platform used for application development.

Device Firmware Update (DFU)
   A mechanism for upgrading the firmware of a device.

Devicetree
   A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, interrupt lines, etc.

Devicetree Specification (DTS, DTSpec)

>    It "provides a complete boot program to client program interface definition, combined with minimum system requirements that facilitate the development of a wide variety of systems." Devicetree Specification, Release v0.3.

Reviewer: `It` > `A specification that`?

Author: Right, fixed.

Device Under Test (DUT)
   A manufactured product undergoing testing.

Docker
   A set of software tools using OS-level virtualization to create and run applications and their dependencies in self-contained environments called containers.

Docker container

>    A self-contained environment created by running a docker container image in the docker engine.

Reviewer: `Docker` capitalized? Looks like this is their way of writing on the docs.docker.com...

Author: Makes sense, it is a trademark

   The control protocol of the IP stack that enables the establishment of reachability, routes, and so on.
   This protocol is an integral part of any IP but is unique as it is not a transport protocol to exchange data between hosts.

Internet Protocol version 4 (IPv4)
   The fourth version in the development of the Internet Protocol (IP).
   It is the communications protocol that provides an identification and location system for computers on networks.
   It routes most traffic on the Internet.

Internet Protocol version 6 (IPv6)
   The latest version of the Internet Protocol (IP).
   It is the communications protocol that provides an identification and location system for computers on networks and routes traffic across the Internet.

IPv4 address
   A numerical label that is used to identify a network interface of a computer or other network node participating in an IPv4 computer network.

IPv6 address
   An alphanumerical label that is used to identify a network interface of a computer or other network node participating in an IPv6 computer network.

Kconfig
   A kernel configuration language.
   It was initially created for the Linux kernel.

Kconfig file

>    A configuration file provided by an application (usually called prj.conf).

Reviewer: ```suggestion
   A configuration file provided by an application (usually called :file:`prj.conf`).
```

   Also known as Cat-M1.

Lightweight Machine to Machine (LwM2M)
   An application layer protocol.
   It defines the service architecture for IoT devices and the protocol for device management.

Man-in-the-Middle (MITM)
   A man-in-the-middle attack is a form of eavesdropping where communication between two devices is monitored and modified by an unauthorized party who relays information between the two devices giving the illusion that they are directly connected.

Manifest file
   The file containing the metadata of the manifest repository.
   In NCS and Zephyr, it is :file:`west.yml`.

Manifest repository
   A repository that contains a :file:`west.yml` file in its root folder and can therefore act as the center of a repository star topology.

Mass Storage Device (MSD)
   Any storage device that makes it possible to store and port large amounts of data in a permanent and machine-readable fashion.

Maximum Transmission Unit (MTU)
   The largest packet or frame that can be sent in a single network-layer transaction.

MCUboot
   A secure bootloader for 32-bit microcontroller units, which is independent of hardware and operating system.

Message Queue Telemetry Transport (MQTT)

>    A machine-to-machine (M2M)/"Internet of Things" connectivity protocol.

Reviewer: ```suggestion
   A protocol in the machine-to-machine (M2M) and Internet of Things connectivity area.
```
Trying to simplify the concatenation of the style here: )/"

Microcontroller Unit (MCU)
   A small computer on an integrated circuit.

Menuconfig
   A tool to configure kernel source.
   It was initially created for the Linux kernel.
   It uses the Kconfig configuration language.

Multicast addressing
   An addressing type that uses a one-to-unique many association, where datagrams are routed from a single sender to multiple selected endpoints simultaneously in a single transmission.

Narrowband Internet of Things (NB-IoT)
   A narrowband technology standard with longer range, lower throughput, and better penetration in, for example, cellars and parking garages compared to LTE-M.
   NB-IoT is most suitable for static, low throughput applications like smart metering, smart agriculture, and smart city applications.
   Also known as Cat-NB1.

Near Field Communication (NFC)
   A standards-based short-range wireless connectivity technology that enables two electronic devices to establish communication by bringing them close to each other.

Network Co-Processor (NCP)
   A co-processor offloading network functions from the host processor.

>    In the |NCS| context, it is tipically used in OpenThread platform designs.

Reviewer: Also Zigbee.
Btw, typo: `t[i]pically`

Network Time Protocol (NTP)
   A networking protocol for clock synchronization between computer systems over packet-switched, variable-latency data networks.

NFC-A Listen Mode
   The initial mode of an NFC Forum Device when it does not generate a carrier.

>    The device listens for the remote field of another device. See Near Field Communication (NFC).

Reviewer: Move the 2nd sentence to the new line.

Noise Factor (NF)
   The relation of the Signal-to-Noise Ratio (SNR) in the device input to the SNR in the device output.

Non-volatile Memory Controller (NVMC)
   A controller used for writing and erasing the internal flash memory and the User Information Configuration Registers (UICR).

nRF Connect for Cloud
   Nordic Semiconductor's platform for connecting IoT devices to the cloud, viewing and analyzing device message data, prototyping ideas that use Nordic Semiconductor chips, and more.
   It includes a public REST API that can be used for building IoT solutions.
   See `nRF Connect for Cloud`_.

nRF repository
   An |NCS| repository that does not have an externally maintained, open-source upstream.
   It is exclusive to Nordic development.

nRF Secure Immutable Bootloader (NSIB)
   A first-stage secure bootloader built and maintained by Nordic Semiconductor.

OpenAMP
   A framework that provides software components that enable the development of software applications for Asymmetric Multiprocessing (AMP) systems.

>    See `OpenAMP`_

Reviewer: ```suggestion
   See `OpenAMP`_.
```

Over-the-Air (OTA)
   Any type of wireless transmission.

Patch
   See Commit.

Power Amplifier (PA)
   A device used to increase the transmit power level of a radio signal.

Power Saving Mode (PSM)
   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
   The device stays dormant during the PSM window.

Printed Circuit Board (PCB)
   A board that connects electronic components.

Protocol Data Unit (PDU)
   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
   PDU mode is one of the two ways of sending and receiving SMS messages.

Pull Request
   A GitHub Pull Request, a set of commits that are sent for code review using GitHub.

Quality of Service (QoS)
   The measured overall performance of a service, such as a telephony or computer network, or a cloud computing service.

Radio Co-Processor (RCP)
   A co-processor offloading radio functions from the host processor.
   In the |NCS| context, it is tipically used in OpenThread platform designs.

Real-time operating system (RTOS)

>    A computing environment that reacts to input within a specific period of time. A real-time deadline can be so small that system reaction appears instantaneous.

Reviewer: Move the 2nd sentence to the new line.

Real-Time Transfer (RTT)
   Proprietary technology for bidirectional communication that supports J-Link devices and ARM-based microcontrollers, developed by SEGGER Microcontroller.

Receive Data (RXD)
   A signal line in a serial interface that receives data from another device.

Repository
   A Git repository in its strict sense, the highest granularity allowed by the Git version control system.

Root of Trust (RoT)
   The property in a secure system that identifies the initial step of a chain of trust.

Sample
   An implementation example that showcases a single feature or library

Secure Access Port Protection (SECUREAPPROTECT)
   A register used to prevent read and write access to all secure CPU registers and secure memory-mapped addresses.
   See Access Port Protection (APPROTECT).

Security Manager Protocol (SMP)
   A protocol used for pairing and key distribution.

SEGGER Embedded Studio (SES)

>    A cross-platform Integrated Development Environment (IDE) for embedded C/C++ programming with support for Nordic Semiconductor devices, produced by SEGGER Microcontroller.

Reviewer: Maybe add: "|NCS| uses a custom Nordic Edition of SES."?

Author: @greg-fer Are you sure this is actually a custom edition of SES? I went through the installation process in a sandbox, earlier today, and the installer mentioned a custom license for users of Nordic products. It was also offering a free test license for general users and a professional license, if you had one. I believe this is because the actual IDE is the same. We should verify this.

Reviewer: It has an NCS section in Options.

Author: I'm pretty sure this depends on the "Nordic Semiconductor nRF CPU Support Package" installed as an additional package. I just managed to get that same section installing the latest "vanilla" SEGGER plus that package. Still, the license management section seems to be different. So I'll just mention the Nordic version available as a zip.

Author: Fixed

   The device listens for the remote field of another device.
   See Near Field Communication (NFC).

Noise Factor (NF)
   The relation of the Signal-to-Noise Ratio (SNR) in the device input to the SNR in the device output.

Non-volatile Memory Controller (NVMC)
   A controller used for writing and erasing the internal flash memory and the User Information Configuration Registers (UICR).

nRF Connect for Cloud
   Nordic Semiconductor's platform for connecting IoT devices to the cloud, viewing and analyzing device message data, prototyping ideas that use Nordic Semiconductor chips, and more.
   It includes a public REST API that can be used for building IoT solutions.
   See `nRF Connect for Cloud`_.

nRF repository
   An |NCS| repository that does not have an externally maintained, open-source upstream.
   It is exclusive to Nordic development.

nRF Secure Immutable Bootloader (NSIB)
   A first-stage secure bootloader built and maintained by Nordic Semiconductor.

OpenAMP
   A framework that provides software components that enable the development of software applications for Asymmetric Multiprocessing (AMP) systems.
   See `OpenAMP`_.

OpenThread

>    A portable and flexible open-source implementation of the Thread networking protocol.

Reviewer: Add a link to openthread.io? (Already in links.txt)

Author: fixed

Board
   In Zephyr and |NCS|, a data structure for describing hardware on the board and its configuration files.

Carrier Wave (CW)
   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.

Cat-M1
   LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.

Cat-NB1
   Narrowband Internet of Things (NB-IoT) User Equipment (UE) category with 200 kHz UE bandwidth and a single RX antenna, specified in 3GPP Release 13.

Chain of Trust
   A sequence of properties identifying the trustworthiness of each layer in the system, all the way back to a property in the system referred to as *root of trust* (RoT).
   A secure system depends on building and maintaining a chain of trust through all the layers in the system.
   Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.

Clone
   A local copy of a remote Git repository obtained with ``git clone``.

Commissioning

>    See Provisioning.

Reviewer 2: Provisioning does not say anything about commissioning. The term needs to be defined.

Author: Actually, this term is not very well defined, even in the specs I consulted for coming up with the definition of provisioning. Often, the meaning of these 2 terms overlap (especially in the NCS docs), and I could not figure out a clear-cut definition of it. @jciupis @kyberdin Do you have any input here?

Reviewer 3: I'm only familiar with the way these terms are used in the Thread specification. There, commissioning is a process of authenticating a new, unknown device and joining it to the network. Provisioning is essentially an exchange of application-specific or vendor-specific data and performing additional configuration based on that. The exact meaning of these two terms may vary depending on the protocol, but I guess the above definitions are a good starting point.

Author: Fixed, in the context of the thread specification.

Commit

>    A Git commit, including a unique SHA and a commit message.

Reviewer 2: Does not really explain what a commit is.

Author: fixed

Commit tag
   A tag prepended to the first line of the commit message to ease filtering and identification of particular commit types.

Contribution
   A change to the codebase sent to a remote repository for inclusion.

Development Kit (DK)
   A hardware development platform used for application development.

Device Firmware Update (DFU)
   A mechanism for upgrading the firmware of a device.

Devicetree

>    A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, interrupt lines, etc.

Reviewer 2: Do not use Latin-based abbreviations -> get rid of etc.

Author: fixed

Docker container
   A self-contained environment created by running a Docker container image in the Docker engine.

Docker container image
   A standard set of binary data that contains an application (or more than one) and all the required dependencies.
   When run on the Docker engine, it creates a Docker container.

Docker engine
   The container runtime that runs Docker images.

Domain Name System (DNS)
   A hierarchical distributed naming system for computers, services, or any resource connected to the Internet or a private network.
   It associates various information with domain names assigned to each of the participating entities.
   Most prominently, it translates domain names, which can be easily memorized by humans, to the numerical IP addresses needed for computer services and devices worldwide.
   The Domain Name System is an essential component of the functionality of most Internet services because it is the Internet's primary directory service.

Downstream
   The repository that is forked off an upstream.

Firmware Over-the-Air (FOTA) update

> 

Reviewer 3: nit: redundant newline

Author: fixed

>    A firmware update done remotely over-the-air (OTA).

Reviewer 4: ```suggestion
   A firmware update performed remotely over-the-air (OTA).
```

GitHub fork
   A `GitHub fork`_ is a copy of a repository inside GitHub, that allows the user to create a Pull Request.

General Packet Radio Services (GPRS)
   Packet-based mobile data service for 2G and 3G mobile networks with data rates of 56-114 kbps/second and continuous connection to the Internet.

General-Purpose Input/Output (GPIO)
   A digital signal pin that can be used as input, output, or both.
   It is uncommitted and controllable by the user at runtime.

Generic Attribute Protocol (GATT)
   "Generic Attribute Profile (GATT) is built on top of the Attribute Protocol (ATT) and establishes common operations and a framework for the data transported and stored by the Attribute Protocol.” Bluetooth Core Specification, Version 5.1, Vol 1, Part A, Section 6.4

Global Navigation Satellite System (GNSS)
   A satellite navigation system with global coverage.
   The system provides signals from space transmitting positioning and timing data to GNSS receivers, which use this data to determine location.

Global Positioning System (GPS)
   A satellite-based radio navigation system that provides its users with accurate location and time information over the globe.

GPIOTE

>    General-Purpose Input/Output Tasks and Events

Reviewer 2: This just opens the abbreviation. Do we have a definition?

Author: fixed

Hardware Flow Control (HWFC)

>    A handshaking mechanism used to prevent an overflow of bytes in modems.

Reviewer 3: This definition is not very well suited to embedded use cases. The purpose of hardware flow control is to avoid buffer overflow in the receiver. The communicating devices can synchronize with each other using RTS and CTS pins, e.g. when the baudrate is high enough for data to appear faster than it can be processed by the receiver. Modems and the RS232 standard don't need to be mentioned at all here

Author: Fixed

Mass Storage Device (MSD)
   Any storage device that makes it possible to store and port large amounts of data in a permanent and machine-readable fashion.

Maximum Transmission Unit (MTU)
   The largest packet or frame that can be sent in a single network-layer transaction.

MCUboot
   A secure bootloader for 32-bit microcontroller units, which is independent of hardware and operating system.

Message Queue Telemetry Transport (MQTT)
   A machine-to-machine (M2M) connectivity protocol used by some IoT devices.
   It is designed as an extremely lightweight publish/subscribe messaging transport.
   It is useful for connections with remote locations where a small code footprint is required and/or network bandwidth is at a premium.
   For example, it has been used in sensors communicating to a broker via satellite link, over occasional dial-up connections with healthcare providers, and in a range of home automation and small device scenarios.

Microcontroller Unit (MCU)
   A small computer on an integrated circuit.

Menuconfig
   A tool to configure kernel source.
   It was initially created for the Linux kernel.
   It uses the Kconfig configuration language.

Multicast addressing

>    An addressing type that uses a one-to-unique many association, where datagrams are routed from a single sender to multiple selected endpoints simultaneously in a single transmission.

Reviewer 3: >one-to-unique many

Can many actually be unique? I think this part can safely be replaced with `one-to-many`

Author: fixed

   It includes a public REST API that can be used for building IoT solutions.
   See `nRF Connect for Cloud`_.

nRF repository
   An |NCS| repository that does not have an externally maintained, open-source upstream.
   It is exclusive to Nordic development.

nRF Secure Immutable Bootloader (NSIB)
   A bootloader created and maintained by Nordic Semiconductor that is built on Chain of Trust architecture.

OpenAMP
   A framework that provides software components that enable the development of software applications for Asymmetric Multiprocessing (AMP) systems.
   See `OpenAMP`_.

OpenThread
   A portable and flexible :ref:`open-source implementation <OpenThread.io>` of the Thread networking protocol.

OpenThread Border Router (OTBR)
   A router that connects a Thread network to other IP-based networks, like Wi-Fi or Ethernet.
   A Thread network requires a Border Router to connect to other networks.

OSS repository
   An |NCS| repository that tracks an upstream Open Source Software counterpart that is externally maintained.

Over-the-Air (OTA)
   Any type of wireless transmission.

Patch

>    See Commit.

Reviewer 2: Commit does not say anything about patch.

Author: fixed

Power Amplifier (PA)
   A device used to increase the transmit power level of a radio signal.

Power Saving Mode (PSM)
   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
   The device stays dormant during the PSM window.

Printed Circuit Board (PCB)
   A board that connects electronic components.

Protocol Data Unit (PDU)
   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
   PDU mode is one of the two ways of sending and receiving SMS messages.

Provisioning
   For a mesh network, this is the process of adding devices to the network.
   For bootloaders, this is the process of storing public key hashes in a separate region of memory from the bootloader image.

Pull Request

>    A GitHub Pull Request, a set of commits that are sent for code review using GitHub.

Reviewer 3: Pull requests are not specific to Github and usually their purpose is not review of the code, but contributing it to a repo

Author: fixed

Quality of Service (QoS)
   The measured overall performance of a service, such as a telephony or computer network, or a cloud computing service.

Radio Co-Processor (RCP)
   A co-processor offloading radio functions from the host processor.
   In the |NCS| context, it is typically used in OpenThread and Zigbee platform designs.

Real-time operating system (RTOS)

>    A computing environment that reacts to input within a specific period of time.

Reviewer 3: I'd replace `computing environment` with `operating system`. If `operating system` requires additional explanations, I'd create a separate entry for it.

Author: fixed

   A real-time deadline can be so small that system reaction appears instantaneous.

Real-Time Transfer (RTT)

>    Proprietary technology for bidirectional communication that supports J-Link devices and ARM-based microcontrollers, developed by SEGGER Microcontroller.

Reviewer 2: ```suggestion
   Proprietary technology for bidirectional communication that supports J-Link devices and Arm-based microcontrollers, developed by SEGGER Microcontroller.
```
This is how the company Arm spells its name nowadays.

Author: fixed.

Receive Data (RXD)
   A signal line in a serial interface that receives data from another device.

Repository

>    A Git repository in its strict sense, the highest granularity allowed by the Git version control system.

Reviewer 3: I don't really understand the part about granularity

Author: fixed

Root of Trust (RoT)
   The property or component in a secure system that provides the foundation of a Chain of Trust.

Sample

>    An implementation example that showcases a single feature or library

Reviewer 3: ```suggestion
   An implementation example that showcases a single feature or library.
```

Author: fixed

Secure Access Port Protection (SECUREAPPROTECT)
   A register used to prevent read and write access to all secure CPU registers and secure memory-mapped addresses.
   See Access Port Protection (APPROTECT).

Security Manager Protocol (SMP)
   A protocol used for pairing and key distribution.

SEGGER Embedded Studio (SES)
   A cross-platform Integrated Development Environment (IDE) for embedded C/C++ programming with support for Nordic Semiconductor devices, produced by SEGGER Microcontroller.
   |NCS| uses a custom :ref:`Nordic Edition of SES <installing_ses>`.

Signal-to-Noise Ratio (SNR)
   The level of signal power compared to the level of noise power, often expressed in decibels (dB).

Simple Network Time Protocol (SNTP)
   A less complex implementation of NTP, using the same protocol but without requiring the storage of state over extended periods of time.

Soft fork
   A fork that contains a very small set of changes when compared to its upstream.

SoftDevice

>    A wireless protocol stack that complements the nRF5 Series SoCs.

Reviewer 3: Isn't SoftDevice strictly a BLE stack?

Author: No worries :D

Transmission Control Protocol (TCP)
   A connection-oriented protocol that provides reliable transport.
   This reliability comes at the cost of control packets overhead of the protocol itself, making it unsuitable for bandwidth-constrained applications.

Toolchain
   A set of development tools.
   See `GNU Arm Embedded Toolchain`_.

Transmit Data (TXD)
   A signal line in a serial interface that transmits data to another device.

Unicast addressing
   An addressing type that uses a one-to-one association between the destination address and the network endpoint.
   Each destination address uniquely identifies a single receiver endpoint.

Universal Asynchronous Receiver/Transmitter (UART)
   A hardware device for asynchronous serial communication between devices.

Universal Serial Bus (USB)
   An industry standard that establishes specifications for cables and connectors and protocols for connection, communication, and power supply between computers, peripheral devices, and other computers.

Upmerge
   The act of updating a downstream repository with a new revision of its upstream counterpart.

Upstream

>    The repository from which a downstream is forked off.

Reviewer 3: ```
Downstream
   The repository that is forked off an upstream.

[...]

Upstream
   The repository from which a downstream is forked off.
```
I see what you did there ;) but one of them should have a precise definition. I'd suggest the upstream

Author: Actually, these definitions were already present in the original version of the glossary, used in the application development section. Btw, you are right, I'll clarify them.

Author: @jciupis Please check if the new definitions (split in branches and repositories) are more complete.

Reviewer 3: I've checked them, they look good to me now :)

   Examples include whether to add networking support, which drivers are needed by the application, etc.

Kconfig language
   A :ref:`kernel configuration language <Kconfig language>` used in Kconfig files and fragments.
   It was initially created for the Linux kernel.

Kconfig option
   A kernel configuration option used in a Kconfig file or fragment.

Kconfig project configuration
   A Kconfig fragment, usually called :file:`prj.conf`, used to define default Kconfig options for an application.
   These are foundational options for the application that will always be applied to its built image.
   However, they can be overridden by applying an additional Kconfig fragment at build time.

Link Layer (LL)
   “A control protocol for the link and physical layers that is carried over logical links in addition to user data.” Bluetooth Core Specification, Version 5.1, Vol 1, Part A, Section 1.2

Low-Noise Amplifier (LNA)
   In a radio receiving system, an electronic amplifier that amplifies a very low-power signal without significantly degrading its signal-to-noise ratio.

LTE-M

>    An open standard that is most suitable for medium-throughput applications requiring low power, low latency, and/or mobility, like asset tracking, wearables, medical, POS, and home security applications.

Reviewer 5: POS could probably be spelled out/defined here

Author: fixed

   A Thread network requires a Border Router to connect to other networks.

Operating System (OS)
   A set of functions and data structures that manages system resources, hardware components, and the execution of programs and processes.
   It is usually composed of a kernel, a scheduler, a file system, a memory manager, and other components.

OSS repository
   An |NCS| repository that tracks an upstream Open Source Software counterpart that is externally maintained.

Over-the-Air (OTA)
   Any type of wireless transmission.

Patch
   See Commit.

Power Amplifier (PA)
   A device used to increase the transmit power level of a radio signal.

Power Saving Mode (PSM)
   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
   The device stays dormant during the PSM window.

Printed Circuit Board (PCB)
   A board that connects electronic components.

Protocol Data Unit (PDU)
   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
   PDU mode is one of the two ways of sending and receiving SMS messages.

Provisioning

>    * In a thread mesh network, the process of associating a device to the appropriate service, and performing any application or vendor-specific configuration.

Reviewer 5: There is a thread mesh network and there is a Bluetooth mesh network.
Provisioning is used by the Bluetooth mesh to add new devices to a mesh network (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/reference/bluetooth/mesh/provisioning.html#bluetooth-mesh-provisioning)

Thread mesh uses comissioning to add devices to a thread network.
See more details here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_thread_commissioning.html

Author: In my last commit I clarified in the context of thread what commissioning and provisioning is.
@mia-ko could you please check the new definitions?

Author: @mia-ko I missed the first question, sorry. At the moment there is no specific rule on what should be added here.

.. _glossary:

Glossary
########

.. contents::
   :local:
   :depth: 2

Anycast addressing
   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
   This is a one-to-nearest association.

Access Port Protection (APPROTECT)
   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.

> API Result

Author: fixed

>    Indicates success or failure of an API procedure.

Author: fixed

   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.

Application

>    An implementation example that includes a variety of libraries to implement a specific use case.

Author: fixed

Application Programming Interface (API)
   A language and message format used by an application program to communicate with an operating system, application, or other services.

Board

>    In Zephyr and the |NCS|, a data structure for describing hardware on the board and its configuration files.

Reviewer 6: We use different boards for non-secure and secure variants of a build, should we mention that here? Currently it seems to map to hardware only.

Author: fixed

Branch
   A line of development composed of a sequence of commits.

Carrier Wave (CW)
   A single-frequency electromagnetic wave that can be modulated in amplitude, frequency, or phase to convey information.

Cat-M1
   LTE-M User Equipment (UE) category with a single RX antenna, specified in 3GPP Release 13.

Cat-NB1
   Narrowband Internet of Things (NB-IoT) User Equipment (UE) category with 200 kHz UE bandwidth and a single RX antenna, specified in 3GPP Release 13.

Chain of Trust
   A sequence of properties identifying the trustworthiness of each layer in the system, all the way back to a property in the system referred to as *root of trust* (RoT).
   A secure system depends on building and maintaining a chain of trust through all the layers in the system.
   Each step in this chain guarantees that the next step can be trusted to have certain properties because any unauthorized modification of a subsequent step will be detected and the process halted.

Clone
   A local copy of a remote Git repository obtained using the command ``git clone``.

Commissioning

>    In a thread mesh network, the process of authenticating and joining new devices to the network.

Author: As discussed yesterday, resolved

Commit
   A snapshot of the history of each project file, taken at a specific moment in time.
   It is associated with a unique SHA and a message describing the edits it contains.

Commit tag
   A tag prepended to the first line of the commit message to ease filtering and identification of particular commit types.

Contribution
   A change to the codebase sent to a remote repository for inclusion.

Development Kit (DK)
   A hardware development platform used for application development.

Device Firmware Update (DFU)
   A mechanism for upgrading the firmware of a device.

Devicetree

>    A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, and interrupt lines.

Reviewer 7: I think data structure is appropriate; it is at least a tree, and spiritually it's a DAG since nodes can point to each other but cycles are disallowed.

>    A data structure for describing hardware and its boot-time configuration, including peripherals on a board, boot-time clock frequencies, and interrupt lines.

Author: fixed

   A manufactured product undergoing testing.

Docker
   A set of software tools using OS-level virtualization to create and run applications and their dependencies in self-contained environments called containers.

Docker container
   A self-contained environment created by running a Docker container image in the Docker engine.

Docker container image
   A standard set of binary data that contains an application (or more than one) and all the required dependencies.
   When run on the Docker engine, it creates a Docker container.

Docker engine
   The container runtime that runs Docker images.

Domain Name System (DNS)
   A hierarchical distributed naming system for computers, services, or any resource connected to the Internet or a private network.
   It associates various information with domain names assigned to each of the participating entities.
   Most prominently, it translates domain names, which can be easily memorized by humans, to the numerical IP addresses needed for computer services and devices worldwide.
   The Domain Name System is an essential component of the functionality of most Internet services because it is the Internet's primary directory service.

> Downstream branch

Author: fixed

   A branch located downstream, relative to another branch, in the flow of information.
   It describes the relationship between branches across remotes.

Downstream repository

>    A repository located downstream, relative to another repository, in the flow of information.

Author: fixed

Firmware Over-the-Air (FOTA) update
   A firmware update performed remotely over-the-air (OTA).

Floating-Point Unit (FPU)
   A part of a CPU specially designed to perform operations on floating-point numbers.

Fork

>    A server-hosted copy of a repository (upstream) that intends to follow the changes made in the original repository as time goes by, while at the same time keeping some other changes unique to it.

Author: fixed

General-Purpose Input/Output (GPIO)
   A digital signal pin that can be used as input, output, or both.
   It is uncommitted and controllable by the user at runtime.

General-Purpose Input/Output Tasks and Events (GPIOTE)
   A module that provides functionality for accessing GPIO pins using tasks and events.
   Each GPIOTE channel can be assigned to one pin.

Generic Attribute Protocol (GATT)
   "Generic Attribute Profile (GATT) is built on top of the Attribute Protocol (ATT) and establishes common operations and a framework for the data transported and stored by the Attribute Protocol.” Bluetooth Core Specification, Version 5.1, Vol 1, Part A, Section 6.4

Global Navigation Satellite System (GNSS)
   A satellite navigation system with global coverage.
   The system provides signals from space transmitting positioning and timing data to GNSS receivers, which use this data to determine location.

Global Positioning System (GPS)
   A satellite-based radio navigation system that provides its users with accurate location and time information over the globe.

Hardware Flow Control (HWFC)

>    A handshaking mechanism used to prevent a buffer overflow in the receiver (in embedded computing use cases).

Author: fixed

   When the transmission baud rate is high enough for data to appear faster than it can be processed by the receiver, the communicating devices can synchronize with each other, using RTS and CTS pins.

Host Controller Interface (HCI)

>    Standardized communication between the host stack (e.g., a PC or mobile phone OS) and the controller (the Bluetooth IC).

Author: fixed

Internet Control Message Protocol (ICMP)
   The control protocol of the IP stack that enables the establishment of reachability, routes, and so on.
   This protocol is an integral part of any IP but is unique as it is not a transport protocol to exchange data between hosts.

Internet Protocol version 4 (IPv4)
   The fourth version in the development of the Internet Protocol (IP).
   It is the communications protocol that provides an identification and location system for computers on networks.
   It routes most traffic on the Internet.

Internet Protocol version 6 (IPv6)
   The latest version of the Internet Protocol (IP).
   It is the communications protocol that provides an identification and location system for computers on networks and routes traffic across the Internet.

IPv4 address
   A numerical label that is used to identify a network interface of a computer or other network node participating in an IPv4 computer network.

IPv6 address
   An alphanumerical label that is used to identify a network interface of a computer or other network node participating in an IPv6 computer network.

Kconfig file

>    A configuration file for a module, written in the Kconfig language syntax.

Reviewer 6: You can also have Kconfig files for samples.

Author: fixed

   It defines build-time configuration options, also called symbols, namely application-specific values for one or more kernel configuration options.
   It also defines how they are grouped into menus and sub-menus, and dependencies between them that determine what configurations are valid.

Kconfig fragment

>    A configuration file used for building an application image with or without software support from specific Kconfig options.

Author: fixed

>    Examples include whether to add networking support, which drivers are needed by the application, etc.

Reviewer 2: We have another etc here. Please get rid of it.

Author: fixed

Kconfig language
   A `kernel configuration language <Kconfig language>`_ used in Kconfig files and fragments.
   It was initially created for the Linux kernel.

Kconfig option

>    A kernel configuration option used in a Kconfig file or fragment.

Author: fixed

Kconfig project configuration
   A Kconfig fragment, usually called :file:`prj.conf`, used to define default Kconfig options for an application.
   These are foundational options for the application that will always be applied to its built image.
   However, they can be overridden by applying an additional Kconfig fragment at build time.

Link Layer (LL)

>    “A control protocol for the link and physical layers that is carried over logical links in addition to user data.” Bluetooth Core Specification, Version 5.1, Vol 1, Part A, Section 1.2

Author: fixed.

Low-Noise Amplifier (LNA)
   In a radio receiving system, an electronic amplifier that amplifies a very low-power signal without significantly degrading its signal-to-noise ratio.

LTE-M
   An open standard that is most suitable for medium-throughput applications requiring low power, low latency, and/or mobility, like asset tracking, wearables, medical, Point of Sale (POS), and home security applications.
   Also known as Cat-M1.

Lightweight Machine to Machine (LwM2M)
   An application layer protocol.
   It defines the service architecture for IoT devices and the protocol for device management.

Man-in-the-Middle (MITM)
   A man-in-the-middle attack is a form of eavesdropping where communication between two devices is monitored and modified by an unauthorized party who relays information between the two devices giving the illusion that they are directly connected.

> Manifest file

Reviewer 6: Should this be prefixed with "West" or "(West)" , a manifest file is used in other situations as well, one being DFU. Same goes for `Manifest repository`

Author: fixed

   A repository that contains a :file:`west.yml` file in its root folder and can therefore act as the center of a repository star topology.

Mass Storage Device (MSD)
   Any storage device that makes it possible to store and port large amounts of data in a permanent and machine-readable fashion.

Maximum Transmission Unit (MTU)
   The largest packet or frame that can be sent in a single network-layer transaction.

MCUboot
   A secure bootloader for 32-bit microcontroller units, which is independent of hardware and operating system.

Message Queue Telemetry Transport (MQTT)
   A machine-to-machine (M2M) connectivity protocol used by some IoT devices.
   It is designed as an extremely lightweight publish/subscribe messaging transport.
   It is useful for connections with remote locations where a small code footprint is required and/or network bandwidth is at a premium.
   For example, it has been used in sensors communicating to a broker via satellite link, over occasional dial-up connections with healthcare providers, and in a range of home automation and small device scenarios.

Microcontroller Unit (MCU)
   A small computer on an integrated circuit.

Menuconfig

>    A tool to configure kernel source.

Reviewer 6: All Kconfig properties are configured through menuconfig. All subsys and samples, not only kernel.

Reviewer 7: I suggest 'A tool to view and edit Kconfig settings'

Author: fixed

Network Co-Processor (NCP)
   A co-processor offloading network functions from the host processor.
   In the |NCS| context, it is typically used in OpenThread and Zigbee platform designs.

Network Time Protocol (NTP)
   A networking protocol for clock synchronization between computer systems over packet-switched, variable-latency data networks.

NFC-A Listen Mode
   The initial mode of an NFC Forum Device when it does not generate a carrier.
   The device listens for the remote field of another device.
   See Near Field Communication (NFC).

Noise Factor (NF)
   The relation of the Signal-to-Noise Ratio (SNR) in the device input to the SNR in the device output.

Non-volatile Memory Controller (NVMC)
   A controller used for writing and erasing the internal flash memory and the User Information Configuration Registers (UICR).

nRF Connect for Cloud
   Nordic Semiconductor's platform for connecting IoT devices to the cloud, viewing and analyzing device message data, prototyping ideas that use Nordic Semiconductor chips, and more.
   It includes a public REST API that can be used for building IoT solutions.
   See `nRF Connect for Cloud`_.

nRF repository

>    An |NCS| repository that does not have an externally maintained, open-source upstream.

Author: fixed

   It is exclusive to Nordic development.

nRF Secure Immutable Bootloader (NSIB)
   A bootloader created and maintained by Nordic Semiconductor that is built on Chain of Trust architecture.

OpenAMP
   A framework that provides software components that enable the development of software applications for Asymmetric Multiprocessing (AMP) systems.
   See `OpenAMP`_.

OpenThread
   A portable and flexible `open-source implementation <OpenThread.io>`_ of the Thread networking protocol.

OpenThread Border Router (OTBR)
   A router that connects a Thread network to other IP-based networks, like Wi-Fi or Ethernet.
   A Thread network requires a Border Router to connect to other networks.

Operating System (OS)
   A set of functions and data structures that manages system resources, hardware components, and the execution of programs and processes.
   It is usually composed of a kernel, a scheduler, a file system, a memory manager, and other components.

OSS repository

>    An |NCS| repository that tracks an upstream Open Source Software counterpart that is externally maintained.

Author: fixed

Over-the-Air (OTA)
   Any type of wireless transmission.

Patch

>    A method to export git commits for sharing them with others.

Reviewer 8: This is also very often used indistinctly with "commit"

Reviewer 7: A patch is not a commit; I think this should read 'A method to describe changes in one or more source code files.'

Author: fixed

   It does not require a repository.

Power Amplifier (PA)
   A device used to increase the transmit power level of a radio signal.

Power Saving Mode (PSM)
   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
   The device stays dormant during the PSM window.

Printed Circuit Board (PCB)
   A board that connects electronic components.

Protocol Data Unit (PDU)
   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
   PDU mode is one of the two ways of sending and receiving SMS messages.

Provisioning
   * In a thread mesh network, the process of associating a device to the appropriate service, and performing any application or vendor-specific configuration.
     It is a step in the commissioning process.
   * In a bluetooth mesh network, the process of adding devices to the network.
   * In a bootloader, the process of storing public key hashes in a separate region of memory from the bootloader image.

Pull Request

>    A set of commits that are sent to contribute to a repository.

Reviewer 8: This only applies in the context of GitHub

Reviewer 3: @carlescufi could you elaborate? This entry used to be Github-specific, but I suggested to make it more general, as the concept of a pull request is widely used, both on Github and outside of it.

Author: @carlescufi

Reviewer 8: OK, I think Pull Request is almost exclusive to GitHub (GitLab, gerrit and others use other terms) but I do think that BitBucket uses it, so I will approve.

Real-time operating system (RTOS)
   An operating system that reacts to input within a specific period of time.
   A real-time deadline can be so small that system reaction appears instantaneous.

Real-Time Transfer (RTT)
   Proprietary technology for bidirectional communication that supports J-Link devices and Arm-based microcontrollers, developed by SEGGER Microcontroller.

Receive Data (RXD)
   A signal line in a serial interface that receives data from another device.

Repository
   The entire set of files and folders of which a project is composed, together with the revision history of each file.
   It is often composed of multiple branches.
   It is also known as *Git repository* or *Git project*.

Root of Trust (RoT)
   The property or component in a secure system that provides the foundation of a Chain of Trust.

Sample
   An implementation example that showcases a single feature or library.

Secure Access Port Protection (SECUREAPPROTECT)
   A register used to prevent read and write access to all secure CPU registers and secure memory-mapped addresses.
   See Access Port Protection (APPROTECT).

> Security Manager Protocol (SMP)

Reviewer 6: There is also the Simple Management Protocol, which unfortunately shares the acronym. Should we explain that as well here to avoid confusion?

Author: fixed

   A less complex implementation of NTP, using the same protocol but without requiring the storage of state over extended periods of time.

Soft fork
   A fork that contains a very small set of changes when compared to its upstream.

SoftDevice
   A wireless protocol stack that complements the nRF5 Series SoCs.
   Nordic Semiconductor provides these stacks as qualified, precompiled binary files.

SoftDevice Manager (SDM)
   A SoftDevice component that controls the SoftDevice state and configures the behavior of certain core functionality.

Software Development Kit (SDK)
   A set of tools used for developing applications for a specific device or operating system.

Spinel
   A general management protocol for enabling a host device to communicate with and manage co-processors, like a network co-processor (NCP) or a radio co-processor (RCP).

System in Package (SiP)
   Several integrated circuits, often from different technologies, enclosed in a single module that performs as a system or subsystem.

System on Chip (SoC)
   A microchip that integrates all the necessary electronic circuits and components of a computer or other electronic systems on a single integrated circuit.

> System Protection Unit (SPU)

Reviewer 6: Isn't the SPU a nordic-only peripheral? The explanation seems right, but maybe we should mention that. As it stands now it looks like it is a general concept.

Author: fixed

Target
   The goal of an operation, for example, programming a specific image on a device, compiling a specific set of files, or removing previously generated files.

Transmission Control Protocol (TCP)
   A connection-oriented protocol that provides reliable transport.
   This reliability comes at the cost of control packets overhead of the protocol itself, making it unsuitable for bandwidth-constrained applications.

Toolchain
   A set of development tools.
   See `GNU Arm Embedded Toolchain`_.

Transmit Data (TXD)
   A signal line in a serial interface that transmits data to another device.

Unicast addressing
   An addressing type that uses a one-to-one association between the destination address and the network endpoint.
   Each destination address uniquely identifies a single receiver endpoint.

Universal Asynchronous Receiver/Transmitter (UART)
   A hardware device for asynchronous serial communication between devices.

Universal Serial Bus (USB)
   An industry standard that establishes specifications for cables and connectors and protocols for connection, communication, and power supply between computers, peripheral devices, and other computers.

Upmerge

>    The act of updating (synchronizing) a downstream repository with a newer revision of its upstream repository.

Reviewer 8: only applies to OSS repositories.

Author: fixed

Upstream branch

>    A branch located upstream, relative to another branch, in the flow of information.

Reviewer 8: I would remove this altogether

Reviewer 3: @carlescufi The idea to create separate entries for upstream branch and upstream repo has resulted from the fact that a single word _upstream_ might mean either repo or branch, depending on the context. Therefore it seemed natural to extend the documentation so that both meanings are covered. Why do you think it's better to remove it?

Reviewer 8: Because usually we do not use branches in our repos. Instead, we use `master` or release branches, which can be translated to "releases".

Reviewer 8: Same as above

Author: fixed

   Upstream repository
      A repository located upstream, relative to another repository, in the flow of information.
      See :ref:`dm_repo_types`.

   User Datagram Protocol (UDP)
      One of the core IP protocols.
      UDP with its connectionless model, no handshaking dialogues makes it a suitable transport for systems with constrained bandwidth like Bluetooth low energy.

   User Information Configuration Registers (UICR)
      Non-volatile memory registers used to configure user-specific settings.

   Watchdog timer (WDT)
      A timer that causes a system reset if it is not poked periodically.

   West
      A multiple repository management system used by Zephyr and the |NCS|.
      See :ref:`zephyr:west`.

   West manifest file
      The file containing the metadata of the manifest repository.
      In the |NCS| and Zephyr, it is :file:`west.yml`.

   West manifest repository
      A repository that contains a :file:`west.yml` file in its root folder and can therefore act as the center of a repository star topology.
      See :ref:`dm_repo_types`.

   West project

>       Any of the listed repositories inside the :file:`west.yml` file in a manifest repository.

Author: fixed

   When run on the Docker engine, it creates a Docker container.

Docker engine
   The container runtime that runs Docker images.

Domain Name System (DNS)
   A hierarchical distributed naming system for computers, services, or any resource connected to the Internet or a private network.
   It associates various information with domain names assigned to each of the participating entities.
   Most prominently, it translates domain names, which can be easily memorized by humans, to the numerical IP addresses needed for computer services and devices worldwide.
   The Domain Name System is an essential component of the functionality of most Internet services because it is the Internet's primary directory service.

Downstream
   The repository that is forked off an upstream.

Floating-Point Unit (FPU)
   A part of a CPU specially designed to perform operations on floating-point numbers.

Fork
   A server-hosted copy of a repository (upstream) that intends to follow the changes made in the original repository as time goes by, while at the same time keeping some other changes unique to it.

> 

Reviewer 4: Perhaps we should add a glossary entry for [FOTA](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/net/fota_download.html?highlight=fota#fota-download)?

Author: There is an entry for Over-the-Air (OTA), but I can also add one for FOTA. My only doubt is if it is actually correct, as OTA could be referred to anything happening over-the-air, while FOTA doesn't make sense unless is "FOTA Update". Maybe I can just add this one.

Author: Added.

Author: Added

   It routes most traffic on the Internet.

Internet Protocol version 6 (IPv6)
   The latest version of the Internet Protocol (IP).
   It is the communications protocol that provides an identification and location system for computers on networks and routes traffic across the Internet.

IPv4 address
   A numerical label that is used to identify a network interface of a computer or other network node participating in an IPv4 computer network.

IPv6 address
   An alphanumerical label that is used to identify a network interface of a computer or other network node participating in an IPv6 computer network.

Kconfig
   A kernel configuration language.
   It was initially created for the Linux kernel.

Kconfig file
   A configuration file provided by an application (usually called :file:`prj.conf` ).
   It describes build-time configuration options, also called symbols, namely application-specific values for one or more kernel configuration options.
   It also defines how they are grouped into menus and sub-menus, and dependencies between them that determine what configurations are valid.
   You can use it to configure software support to build into the final image.

>    Examples include whether to add networking support, which drivers are needed by the application, etc.

Reviewer 4: This glossary entry is combining a couple things that should probably be separated.

A _Kconfig file_ is technically the `Kconfig` files found in various folders throughout NCS and Zephyr that define the configurations, options, etc. for a module, and are written in the [Kconfig language](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html) syntax. Lines 169 and 170 would be relevant for this description.

A _Kconfig fragment_ is a combination of lines 168, 171, and 172. Perhaps recombined into:

```
A configuration file used for building an application image with or without software support from specific Kconfig options.
Examples include whether to add networking support, which drivers are needed by the application, etc.
```

And maybe add a _Kconfig project configuration_ (or just _Project configuration_?) with a brief summary:

```
A Kconfig fragment used to define default Kconfig options for an application (usually called :file:`prj.conf`).
These are foundational options for the application that will always be applied to its built image.
However, they can be overridden by applying an additional Kconfig fragment at build time.
```

Author: Fixed.

Network Time Protocol (NTP)
   A networking protocol for clock synchronization between computer systems over packet-switched, variable-latency data networks.

NFC-A Listen Mode
   The initial mode of an NFC Forum Device when it does not generate a carrier.
   The device listens for the remote field of another device.
   See Near Field Communication (NFC).

Noise Factor (NF)
   The relation of the Signal-to-Noise Ratio (SNR) in the device input to the SNR in the device output.

Non-volatile Memory Controller (NVMC)
   A controller used for writing and erasing the internal flash memory and the User Information Configuration Registers (UICR).

nRF Connect for Cloud
   Nordic Semiconductor's platform for connecting IoT devices to the cloud, viewing and analyzing device message data, prototyping ideas that use Nordic Semiconductor chips, and more.
   It includes a public REST API that can be used for building IoT solutions.
   See `nRF Connect for Cloud`_.

nRF repository
   An |NCS| repository that does not have an externally maintained, open-source upstream.
   It is exclusive to Nordic development.

nRF Secure Immutable Bootloader (NSIB)

>    A first-stage secure bootloader built and maintained by Nordic Semiconductor.

Reviewer 4: ```suggestion
   A bootloader created and maintained by Nordic Semiconductor that is built on Chain of Trust architecture.
```

Author: fixed

   See `OpenAMP`_.

OpenThread
   A portable and flexible open-source implementation of the Thread networking protocol.

OpenThread Border Router (OTBR)
   A router that connects a Thread network to other IP-based networks, like Wi-Fi or Ethernet.
   A Thread network requires a Border Router to connect to other networks.

OSS repository
   An |NCS| repository that tracks an upstream Open Source Software counterpart that is externally maintained.

Over-the-Air (OTA)
   Any type of wireless transmission.

Patch
   See Commit.

Power Amplifier (PA)
   A device used to increase the transmit power level of a radio signal.

Power Saving Mode (PSM)
   A feature introduced in 3GPP Release 12 to improve battery life of IoT (Internet of Things) devices by minimizing energy consumption.
   The device stays dormant during the PSM window.

Printed Circuit Board (PCB)
   A board that connects electronic components.

Protocol Data Unit (PDU)
   Information transferred as a single unit between peer entities of a computer network and containing control and address information or data.
   PDU mode is one of the two ways of sending and receiving SMS messages.

Provisioning

>    The process of adding devices to a mesh network.

Reviewer 4: ```suggestion
   For a mesh network, this is the process of adding devices to the network.
   For bootloaders, this is the process of storing public key hashes in a separate region of memory from the bootloader image.
```

Author: fixed

Pull Request
   A GitHub Pull Request, a set of commits that are sent for code review using GitHub.

Quality of Service (QoS)
   The measured overall performance of a service, such as a telephony or computer network, or a cloud computing service.

Radio Co-Processor (RCP)
   A co-processor offloading radio functions from the host processor.
   In the |NCS| context, it is typically used in OpenThread and Zigbee platform designs.

Real-time operating system (RTOS)
   A computing environment that reacts to input within a specific period of time.
   A real-time deadline can be so small that system reaction appears instantaneous.

Real-Time Transfer (RTT)
   Proprietary technology for bidirectional communication that supports J-Link devices and ARM-based microcontrollers, developed by SEGGER Microcontroller.

Receive Data (RXD)
   A signal line in a serial interface that receives data from another device.

Repository
   A Git repository in its strict sense, the highest granularity allowed by the Git version control system.

Root of Trust (RoT)

>    The property in a secure system that identifies the initial step of a chain of trust.

Reviewer 4: ```suggestion
   The property or component in a secure system that provides the foundation of a Chain of Trust.
```

Author: fixed
